pages/_app.js

```diff
@@ -2,16 +2,24 @@ import GlobalStyle from "@/styles";
 import Head from "next/head";
 import { SWRConfig } from "swr";
 
-const fetcher = (url) => fetch(url).then((response) => response.json());
-
 export default function App({ Component, pageProps }) {
   return (
     <>
-      <GlobalStyle />
-      <Head>
-        <title>Capstone Project</title>
-      </Head>
-      <SWRConfig value={{ fetcher }}>
+      <SWRConfig
+        value={{
+          fetcher: async (...args) => {
+            const response = await fetch(...args);
+            if (!response.ok) {
+              throw new Error(`Request with ${JSON.stringify(args)} failed.`);
+            }
+            return await response.json();
+          },
+        }}
+      >
```

Comment:
I would still define the fetcher outside of the `App` component:

```js
async function fetcher(...args) {
  const response = await fetch(...args);
  if (!response.ok) {
    throw new Error(`Request with ${JSON.stringify(args)} failed.`);
  }
  return await response.json();
}

export default function App({ Component, pageProps }) {
  return <SWRConfig value={fetcher}>
    ...
  </SWRConfig>;
}
```

Reply:
You are absolutely right! So I can use the function in other multiple places